File: push/src/main/java/com/nihility/Global.java
package com.nihility;

import com.nihility.service.RegistrationRecorder;
import com.nihility.utils.Singleton;
import com.xiaomi.xmsf.push.utils.ConfigValueConverter;
import com.xiaomi.xmsf.push.utils.IconConfigurations;
import com.xiaomi.xmsf.utils.ConfigCenter;

import top.trumeet.common.cache.ApplicationNameCache;
import top.trumeet.common.cache.IconCache;

public class Global {
    public static MethodHooker MethodHooker() {
        return Singleton.instance();
    }

    public static void setMethodHooker(MethodHooker methodHooker) {
        Singleton.reset(methodHooker);
    }

    public static MiPushEventListener MiPushEventListener() {
        return Singleton.instance();
    }

    public static void setMiPushEventListener(MiPushEventListener listener) {
        Singleton.reset(listener);
    }

    public static RegistrationRecorder RegistrationRecorder() {
        return Singleton.instance();
    }

    public static void setRegistrationRecorder(RegistrationRecorder instance) {
        Singleton.reset(instance);
    }

    public static ConfigValueConverter ConfigValueConverter() {
        return Singleton.instance();
    }

    public static IconConfigurations IconConfigurations() {
        return Singleton.instance();
    }

    public static ConfigCenter ConfigCenter() {
        return Singleton.instance();
    }

    public static void setConfigCenter(ConfigCenter configCenter) {
        Singleton.reset(configCenter);
    }

    public static ApplicationNameCache ApplicationNameCache() {
        return Singleton.instance();
    }

    public static IconCache IconCache() {
        return Singleton.instance();
    }
}

File: push/src/main/java/com/nihility/Global.kt
@file:Suppress("DEPRECATION")
package com.nihility
// Compatibility shim: legacy namespace forwarding to com.magisk317.*

import com.magisk317.service.RegistrationRecorder
import com.xiaomi.xmsf.push.utils.ConfigValueConverter
import com.xiaomi.xmsf.push.utils.IconConfigurations
import com.xiaomi.xmsf.utils.ConfigCenter
import top.trumeet.common.cache.ApplicationNameCache
import top.trumeet.common.cache.IconCache

@Deprecated(
    message = "Use com.magisk317.Global instead.",
    replaceWith = ReplaceWith("com.magisk317.Global")
)
object Global {
    @JvmStatic
    fun HookHandler(): HookHandler = com.magisk317.Global.HookHandler()

    @JvmStatic
    fun setHookHandler(hookHandler: HookHandler) {
        com.magisk317.Global.setHookHandler(hookHandler)
    }

    @JvmStatic
    fun MiPushEventListener(): MiPushEventListener = com.magisk317.Global.MiPushEventListener()

    @JvmStatic
    fun setMiPushEventListener(listener: MiPushEventListener) {
        com.magisk317.Global.setMiPushEventListener(listener)
    }

    @JvmStatic
    fun RegistrationRecorder(): RegistrationRecorder = com.magisk317.Global.RegistrationRecorder()

    @JvmStatic
    fun setRegistrationRecorder(instance: RegistrationRecorder) {
        com.magisk317.Global.setRegistrationRecorder(instance)
    }

    @JvmStatic
    fun ConfigValueConverter(): ConfigValueConverter = com.magisk317.Global.ConfigValueConverter()

    @JvmStatic
    fun IconConfigurations(): IconConfigurations = com.magisk317.Global.IconConfigurations()

    @JvmStatic
    fun ConfigCenter(): ConfigCenter = com.magisk317.Global.ConfigCenter()

    @JvmStatic
    fun setConfigCenter(configCenter: ConfigCenter) {
        com.magisk317.Global.setConfigCenter(configCenter)
    }

    @JvmStatic
    fun ApplicationNameCache(): ApplicationNameCache = com.magisk317.Global.ApplicationNameCache()

    @JvmStatic
    fun IconCache(): IconCache = com.magisk317.Global.IconCache()
}
